DqtNotifications/terraform/app.tf

```diff
@@ -0,0 +1,153 @@
+locals {
+  auth_rule_name = "SendAndListenSharedAccessKey"
+  sku            = var.enable_private_endpoint == true ? "Premium" : var.sku
+  capacity       = local.sku == "Premium" && var.capacity <= 0 ? 1 : var.capacity
+       }
+
+# App Sercie Plan
+resource "azurerm_app_service_plan" "app_service_plan" {
+  name                = local.app_service_plan_name
+  location            = data.azurerm_resource_group.rgsb.location
+  resource_group_name = data.azurerm_resource_group.rgsb.name
+  kind                = "Linux"
+  reserved            = true
+
+  sku {
+    tier = "Dynamic"
```

Comment:
@gunndabad a [consumption plan](https://learn.microsoft.com/en-us/azure/azure-functions/consumption-plan) is the most cost-effective choice but may not be appropriate depending on how this app will be used.  I think I remember functions being slow to start up after periods of idleness when we tried this in NCS.  Any thoughts?

Reply:
This will likely be running almost all of the time so perhaps a non-consumption plan would be better.

Reply:
👍 @enformatique11 can you make the tier and size a variable so we can pass in different values for each environment via tfvars